base/mpfr.jl

```diff
@@ -39,8 +39,47 @@ function __init__()
     nothing
 end
 
-const ROUNDING_MODE = Ref{Cint}(0) # actually an Enum, defined by `to_mpfr`
-const DEFAULT_PRECISION = Ref{Int}(256)
+"""
+    MPFR.MPFRRoundingMode
+
+Matches the `mpfr_rnd_t` enum provided by MPFR, see
+https://www.mpfr.org/mpfr-current/mpfr.html#Rounding-Modes
+
+This is for internal use, and ensures that `ROUNDING_MODE[]` is type-stable.
+"""
+@enum MPFRRoundingMode begin
+    MPFRRoundNearest
+    MPFRRoundToZero
+    MPFRRoundUp
+    MPFRRoundDown
+    MPFRRoundFromZero
+    MPFRRoundFaithful
+end
+
+convert(::Type{MPFRRoundingMode}, ::RoundingMode{:Nearest})  = MPFRRoundNearest
+convert(::Type{MPFRRoundingMode}, ::RoundingMode{:ToZero})   = MPFRRoundToZero
+convert(::Type{MPFRRoundingMode}, ::RoundingMode{:Up})       = MPFRRoundUp
+convert(::Type{MPFRRoundingMode}, ::RoundingMode{:Down})     = MPFRRoundDown
+convert(::Type{MPFRRoundingMode}, ::RoundingMode{:FromZero}) = MPFRRoundFromZero
+
+function convert(::Type{RoundingMode}, r::MPFRRoundingMode)
+    if r == MPFRRoundNearest
```

Comment:
I don't know if this can lead to noticeable performance difference, but did you consider something like `return (RoundNearest, RoundToZero, ...)[Int(r)+1]` ? (after checking out-of-bounds `r`).